corelib-definitions/src/main/java/eu/europeana/corelib/definitions/model/facets/logic/ImageTagExtractor.java

```diff
@@ -0,0 +1,261 @@
+package eu.europeana.corelib.definitions.model.facets.logic;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+
+/**
+ * Extracts the pure tags from an image resource and generates the fake tags.
+ */
+public class ImageTagExtractor {
+
+    private static final Logger LOG = LogManager.getLogger(ImageTagExtractor.class.getName());
+
+    private final static HashMap<String, Integer> hexColors = new HashMap<>();
```

Comment:
Performance wise it's better to always state a minimum size of you know a map is gonna be big. In this case you actually know the exact size you need, give that is the only parameter of the HashMap constructor.
.... = new HashMap<>(339);